### PR TITLE
serialization RpcException bug pr rebase

### PR DESCRIPTION
**Pull request by wxbty**
## What is the purpose of the change
rebase  https://github.com/apache/dubbo/pull/12280 because of merged by branch 2.7

### CODE REVIEW

**Comment by AlbumenJ** [comment]
Please fix uts

**Comment by AlbumenJ** [comment]
How about change it to judge IOException? Let's standardize all exceptions thrown under Serialzation into IOException, and then unify catch IOException to judge whether it is serialization. In otherwise, such transform will discard some real network exception.

**Comment by wxbty** [comment]
> How about change it to judge IOException? Let's standardize all exceptions thrown under Serialzation into IOException, and then unify catch IOException to judge whether it is serialization. In otherwise, such transform will discard some real network exception.

PTAL again

File: dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultSerializationExceptionWrapper.java [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        }
+    }
+
+    private static Throwable handleToIOException(Exception e) {
+        Throwable t = e.getCause();
+        if (!(t instanceof IOException)) {
+            t = new IOException(new SerializationException(e));
+        }
+        return t;
+    }
```
Comment (AlbumenJ) [inline dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultSerializationExceptionWrapper.java:363]:
```suggestion
    private static IOException handleToIOException(Exception e) {
        if (!(e instanceof IOException)) {
            return new IOException(new SerializationException(e));
        }
        return (IOException) e;
    }
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/DeprecatedExchangeCodec.java [hunk 2]
```diff
@@ -282,7 +282,7 @@ protected void encodeResponse(Channel channel, OutputStream os, Response res) th
                     logger.warn(TRANSPORT_FAILED_RESPONSE, "", "", "Fail to encode response: " + res + ", send bad_response info instead, cause: " + t.getMessage(), t);
 
                     Response r = new Response(res.getId(), res.getVersion());
-                    r.setStatus(Response.BAD_RESPONSE);
+                    r.setStatus(Response.SERIALIZATION_ERROR);
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/DeprecatedExchangeCodec.java:285]: Check if `t` is instance of `IOException`

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/DeprecatedExchangeCodec.java (hunk 2 above)
Comment (wxbty) [reply dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/DeprecatedExchangeCodec.java]: done

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**